main/src/ca/uwaterloo/flix/api/lsp/provider/CompletionProvider.scala

```diff
@@ -490,15 +503,22 @@ object CompletionProvider {
   private def getContext(source: String, uri: String, pos: Position): Option[Context] = {
       val x = pos.character - 1
       val y = pos.line - 1
-      for(line <- source.linesWithSeparators.slice(y, y + 1).toList.headOption) yield {
+      val lines = source.linesWithSeparators.toList
+      for(line <- lines.slice(y, y + 1).toList.headOption) yield {
         val (prefix, suffix) = line.splitAt(x)
         val reversedPrefix = prefix.reverse
         val wordStart = reversedPrefix.takeWhile(isWordChar).reverse
         val wordEnd = suffix.takeWhile(isWordChar)
         val word = wordStart + wordEnd
         val start = x - wordStart.length
         val end = x + wordEnd.length
-        val previousWord = reversedPrefix.dropWhile(isWordChar).dropWhile(_.isWhitespace).takeWhile(isWordChar).reverse
+        val prevWord = reversedPrefix.dropWhile(isWordChar).dropWhile(_.isWhitespace).takeWhile(isWordChar).reverse
+        val previousWord = if (prevWord.nonEmpty) {
+          prevWord
+        } else lines.slice(y - 1, y).toList.headOption match {
```

Comment:
Could this slice y - 1 every be out of bounds?

Reply:
Yes, it could. But in that case it returns an empty list, and `headOption` therefore returns `None`, so everything should work correctly I believe.